### PR TITLE
feat(TFD-1696) new Type Converter component

### PR DESCRIPTION
**Pull request by lbourgeois**
**What is the current behavior?** (You can also link to an open issue here)
In Data Streams we need to be able to convert fields of records into specific, strong data type but can't do it

**What is the new behavior?**
Adding a new component TypeConverter to fill this need

**Please check if the PR fulfills these requirements**

- [X] The commit message follows Talend standard
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features) ?
- [ ] The code coverage on new code >75%
- [ ] The new code does not introduce new technical issues (sonar / eslint)

**What kind of change does this PR introduce?**

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build / CI related changes
- [ ] Other... Please describe:

**Does this PR introduce a breaking change?**

- [ ] Yes
- [X] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/1250/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/1250/)

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/1280/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/1280/)

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/1282/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/1282/)

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/1281/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/1281/)

File: components/components-processing/processing-definition/src/main/java/org/talend/components/processing/definition/typeconverter/TypeConverterProperties.java [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+    }
+
+    @Override
+    public void setupProperties() {
+        super.setupProperties();
+        setupLayout();
+        // Add a default converter
+        converters.createAndAddRow();
+    }
+}
```
Comment (pbailly) [inline components/components-processing/processing-definition/src/main/java/org/talend/components/processing/definition/typeconverter/TypeConverterProperties.java]: I don't want to look too much like another person in the team, but could you add a line return at the end of each file?

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/1283/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/1283/)

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/1284/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/1284/)

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/1285/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/1285/)

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/1287/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/1287/)

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/1288/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/1288/)

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/1351/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/1351/)

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/1352/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/1352/)

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/1354/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/1354/)

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/1355/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/1355/)

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/1358/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/1358/)

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/components_pull-request_master/1360/) [:star:](https://ci-common.datapwn.com/job/components_pull-request_master/1360/) 
[Sonar results](https://sonar.datapwn.com/overview?id=org.talend.components%3Acomponents%3Awip%2Frskraba%2FTFD-1696-type-converter)

**Pull request merged**